sangryeol/Week_6/기둥과 보 설치.java

```diff
@@ -0,0 +1,67 @@
+import java.util.*;
+
+public class Programmers_기둥과보 {
+    class Struct implements Comparable<Struct> {
+        int x, y, a;
+        public Struct(int x, int y, int a) {
+            this.x = x;
+            this.y = y;
+            this.a = a;
+        }
+        public boolean isColumn() {
+            return a == 0;
+        }
+        public boolean hasBaseColumn(TreeSet<Struct> cache) {
+            return y == 0 ||
+                    cache.contains(new Struct(x, y - 1, 0)) ||
+                    cache.contains(new Struct(x, y, 1)) ||
+                    cache.contains(new Struct(x - 1, y, 1));
+        }
+        public boolean hasBaseBo(TreeSet<Struct> cache) {
+            return cache.contains(new Struct(x, y - 1, 0)) ||
+                    cache.contains(new Struct(x + 1, y - 1, 0)) ||
+                    (cache.contains(new Struct(x - 1, y, 1)) && cache.contains(new Struct(x + 1, y, 1)));
+        }
+        public int[] getArray() {
+            return new int[]{x, y, a};
+        }
+        @Override
+        public int compareTo(Struct o) {
```

Comment:
https://stackoverflow.com/questions/25198475/treeset-contains-does-not-call-overwritten-equals